kernel/src/expressions/mod.rs

```diff
@@ -125,6 +242,14 @@ pub enum Expression {
     Struct(Vec<Expression>),
     /// An expression that takes two expressions as input.
     Binary(BinaryExpression),
+    /// An opaque expression (ie defined and implemented by the engine). Kernel does not attempt to
+    /// interpret the expression at all, but will preserve it so the engine can still evaluate it.
+    Opaque(OpaqueExpression),
+    /// An unknown predicate (i.e. one that neither kernel nor engine attempts to evaluate). Kernel
+    /// refuses to evaluate expression trees containing unknown expressions, and engines should do
```

Comment:
nit: We will evaluate expressions with `Unknown` if they appear in an AND. So maybe something like:
"Kernel will not evaluate expression trees rooted at an unknown expression. Note that an `Unknown` expression can appear in an `AND` conjunction, in which case it will simply be ignored for the purposes of data skipping"